src/d/actor/d_a_npc_km1.cpp

```diff
@@ -4,248 +4,816 @@
+    gravity = -4.0;
+    field_0x798 = current.pos;
+    set_action(&daNpc_Km1_c::wait_action1,NULL);
+    shape_angle = current.angle;
+    mStts.Init(0xFF,0xFF,this);
+    mCyl.SetStts(&mStts);
+    mCyl.Set(dNpc_cyl_src);
+    mpMorf->setMorf(0.0);
+    field_0x7BC = 1;
+    setMtx();
+    return 1;
+
 }
 
 /* 0000043C-0000054C       .text setMtx__11daNpc_Km1_cFv */
 void daNpc_Km1_c::setMtx() {
-    /* Nonmatching */
+
+ u32 uVar1;
+ char cVar2;
+ u8 bVar3;
+ 
+    if (field_0x7C7 == 0) {
+    playTexPatternAnm();
+    uVar1 = mpMorf->play(&eyePos,0,0);
+    field_0x7B4 = uVar1;
+    if (mpMorf->getFrame() < field_0x7A4) {
+        field_0x7B4 = 1;
+    }
+    field_0x7A4 = mpMorf->getFrame();
+    mObjAcch.CrrPos(*dComIfG_Bgsp());
+    ;
+    }
+    cVar2 = dComIfG_Bgsp()->GetRoomId(mObjAcch.m_gnd);
+    tevStr.mRoomNo = cVar2;
+    bVar3 = dComIfG_Bgsp()->GetPolyColor(mObjAcch.m_gnd);
+    tevStr.mEnvrIdxOverride = bVar3;
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mpMorf->getModel()->setBaseTRMtx(mDoMtx_stack_c::get());
+    mpMorf->calc();
+    setAttention();
 }
 
+
 /* 0000054C-00000654       .text anmResID__11daNpc_Km1_cFiPiPi */
-void daNpc_Km1_c::anmResID(int, int*, int*) {
-    /* Nonmatching */
+bool daNpc_Km1_c::anmResID(int i_num, int* o_bck_num, int* o_bas_num) {
+    const int ANM_END = 1;
+    static const int a_anm_idx_tbl[1][2] = {1,0};
+
+    JUT_ASSERT(0x11A,0 <= i_num && i_num < ANM_END);
+    JUT_ASSERT(0x11B,o_bck_num && o_bas_num);
+    *o_bck_num = a_anm_idx_tbl[i_num][0];
+    *o_bas_num = a_anm_idx_tbl[i_num][1];
+    return 1;
 }
 
+
 /* 00000654-000006F0       .text BtpNum2ResID__11daNpc_Km1_cFiPi */
-void daNpc_Km1_c::BtpNum2ResID(int, int*) {
-    /* Nonmatching */
+void daNpc_Km1_c::BtpNum2ResID(int i_num, int* o_btp_num)
+{
+    const int TEXPATTERN_END = 1;
+    static const int a_btp_arc_ix_tbl[1] = {3};
+    JUT_ASSERT(0x130,0 <= i_num && i_num < TEXPATTERN_END);
+    *o_btp_num = a_btp_arc_ix_tbl[i_num];
+        return;
 }
 
 /* 000006F0-00000734       .text setAnm_tex__11daNpc_Km1_cFSc */
-void daNpc_Km1_c::setAnm_tex(signed char) {
-    /* Nonmatching */
+void daNpc_Km1_c::setAnm_tex(signed char i_param_1) {
+    if(i_param_1 >= 0 && i_param_1 != field_0x7CD){
+        field_0x7CD = i_param_1;
+        initTexPatternAnm(true);
+    }
+    return;
 }
 
 /* 00000734-00000850       .text init_btp__11daNpc_Km1_cFbi */
-void daNpc_Km1_c::init_btp(bool, int) {
-    /* Nonmatching */
+u8 daNpc_Km1_c::init_btp(bool param_1, int param_2) {
+
+ int btp_num;
+ int iVar3;
+ J3DModelData *pJVar4;
+
+ pJVar4 = mpMorf->getModel()->getModelData();
+ if (param_2 >= 0) {
+   BtpNum2ResID(param_2,&btp_num);
+    J3DAnmTexPattern* pVVar1 = (J3DAnmTexPattern*)dComIfG_getObjectIDRes("Km",btp_num);
+    m_head_tex_pattern = pVVar1;
+   JUT_ASSERT(0x151,m_head_tex_pattern != 0);
+   iVar3 = mBtpAnm.init(pJVar4,m_head_tex_pattern,1,2,1.0,0,-1,param_1,0);
```

Comment:
```suggestion
   iVar3 = mBtpAnm.init(pJVar4, m_head_tex_pattern, TRUE, J3DFrameCtrl::EMode_LOOP, 1.0f, 0, -1, param_1, FALSE);
```
Missing boolean types, loopmode enum, and `f` float suffix